.github/actions/setup-opensearch-dashboards/action.yml

```diff
@@ -0,0 +1,269 @@
+name: 'Starts Dashboards with Plugin'
+description: 'Installs OpenSearch Dashboard with a plugin from github, then checks out the correct dashboards version for the plugin, configures npm/yarn, bootstraps Dashboards, and starts the instance'
+
+inputs:
+  plugin_name:
+    description: 'The name of the plugin to use, such as security-dashboards-plugin'
+    required: true
+
+  built_plugin_name:
+    description: 'The name of the plugin after building, if doing zip install'
+    required: false
+
+  built_plugin_suffix:
+    description: 'The suffix of the plugin after building, if doing zip instll. Leave empty to indicate default OSD build behavior'
+    required: false
+
+  install_zip:
+    description: 'Whether the setup should be done using the install plugin flow. Leave empty to indicate dev setup. Only applicable for linux for now'
+    required: false
+
+  plugin_repo:
+    description: 'The repo of the plugin to use. Leave empty to indicate the repo which is running this action'
+    required: false
+
+  opensearch_dashboards_yml:
+    description: 'The file to replace opensearch_dashboards.yml. Leave empty to use the default'
+    required: false
+
+  plugin_branch:
+    description: 'The branch of the repo of the plugin to use. Leave empty to indicate the branch which is running this action'
+    required: false
+
+outputs:
+  dashboards-directory:
+    description: "The directory where the dashboards has been configured"
+    value: ${{ steps.determine-dashboards-directory.outputs.dashboards-directory }}
+
+  dashboards-binary-directory:
+    description: "The directory of the dashboards binary that was configured"
+    value: ${{ steps.determine-dashboards-directory-zip-install.outputs.dashboards-directory }}
+
+  plugin-directory:
+    description: "The directory where the plugin has been configured"
+    value: ${{ steps.determine-plugin-directory.outputs.plugin-directory }}
+
+
+runs:
+  using: "composite"
+  steps:
+    - id: determine-dashboards-directory
+      if: ${{ inputs.install_zip != 'true' }}
+      run: echo "dashboards-directory=OpenSearch-Dashboards"  >> $GITHUB_OUTPUT
+      shell: bash
+
+    - id: determine-plugin-directory
+      run: echo "::set-output name=plugin-directory::./OpenSearch-Dashboards/plugins/${{ inputs.plugin_name }}"
+      shell: bash
+
+    - uses: actions/checkout@v2
+      with:
+        path: OpenSearch-Dashboards
+        repository: cwperks/OpenSearch-Dashboards
+        ref: 'support-repeated-plugin'
+        fetch-depth: 0
+        filter: |
+          cypress
+          test
+
+    - uses: actions/checkout@v2
+      with:
+        path: ${{ steps.determine-plugin-directory.outputs.plugin-directory }}
+
+    - uses: actions/checkout@v2
+      with:
+        path: ./OpenSearch-Dashboards/plugins/security-admin-dashboards-plugin
```

Comment:
Attach to the above question: is this the second installation for `Security Admin Pages`?